pkg/server/image_pull.go

```diff
@@ -425,9 +425,9 @@ func newTransport() *http.Transport {
 	return &http.Transport{
 		Proxy: http.ProxyFromEnvironment,
 		DialContext: (&net.Dialer{
-			Timeout:   30 * time.Second,
-			KeepAlive: 30 * time.Second,
-			DualStack: true,
+			Timeout:       30 * time.Second,
+			KeepAlive:     30 * time.Second,
+			FallbackDelay: 300 * time.Millisecond,
```

Comment:
fair to remove the dual stack setting since it's now true by default, but why introduce an explicit constant for the 0 maps to 300mil fallback delay?  Would personally prefer `// TODO () need a config option for fallbackdelay default is zero which is mapped to 300mil by fallbackDelay() in golang net/dial.go`

Reply:
I wasn't sure if we _did_ want to have an option for this, hence kept it in line with the other options which set a default here (and make sure that possible updates in Golang's defaults wouldn't introduce a change in behaviour). Happy to remove it though if you prefer that.

As to adding a "TODO"; I see there's an existing TODO on the function itself;

```
TODO(random-liu): Create a library and share this code with `ctr`.
```

I _think_ it's worth looking at that (as a follow-up); there's some work done in containerd to provide a package for pulling images (https://github.com/containerd/containerd/pull/3728); possibly look at what's missing and add those parts, then use it in this repository